### PR TITLE
Add requirement methods to LeafContext

### PR DESCRIPTION
**Pull request by tonyarnold**
Adds methods to `LeafContext` to verify the existence, or non-existence of a context body, as well as to verify the context contains a specific number of parameters.

### CODE REVIEW

**Comment by tonyarnold** [comment]
I've had to strip all of the stack trace and source location information from the `LeafKitError` type - although to be honest, I'm not sure that this information was necessarily _useful_, given that it didn't resolve to locations inside the templates themselves. 

There's room for improvement here.

**Comment by tonyarnold** [comment]
Belay that, I noticed everything else is throwing descriptive strings: let's just use that for now.

File: Sources/LeafKit/LeafRenderer.swift [hunk 1]
```diff
@@ -73,6 +73,29 @@ public struct LeafContext {
         self.body = body
         self.userInfo = userInfo
     }
+
+    /// Throws an error if the parameter count does not equal the supplied number `n`.
+    public func requireParameterCount(_ n: Int) throws {
+        guard parameters.count == n else {
+            throw "Invalid parameter count: \(parameters.count)/\(n)"
```
Comment (tanner0101) [inline Sources/LeafKit/LeafRenderer.swift]: We should really fix this eventually >_<.

**Comment by tanner0101** [comment]
These changes are now available in [1.0.0-rc.1.3](https://github.com/vapor/leaf-kit/releases/tag/1.0.0-rc.1.3)

**Pull request merged**